lib/failover_ip.rb

```diff
@@ -70,18 +78,25 @@ def switch_ips
 
       old_target = current_target
 
-      Hooks.run_before failover_ip, old_target, new_ip[:target]
+      Hooks.run_before failover_ip, old_target, new_ip[:target], dry
 
-      raise unless HTTParty.post("#{base_url}/failover/#{failover_ip}", :body => { :active_server_ip => new_ip[:target] }, :basic_auth => basic_auth).success?
+      if !dry
+        response = HTTParty.post("#{base_url}/failover/#{failover_ip}", :body => { :active_server_ip => new_ip[:target] }, :basic_auth => basic_auth)
+        raise unless response.success?
+        $logger.info "Switch #{failover_ip} to #{new_ip[:target]} completed"
+      else
+        $logger.info "Dry run: would have switched #{failover_ip} to #{new_ip[:target]}"
+      end
 
-      Hooks.run_after failover_ip, old_target, new_ip[:target]
+      Hooks.run_after failover_ip, old_target, new_ip[:target], dry
 
       return true
     end
 
     false
   rescue
-    $logger.error "Unable to set a new active server ip for #{failover_ip}"
+    $logger.error "Unable to set a new active server ip for #{failover_ip} via POST to #{base_url}/failover/#{failover_ip}, :body => { :active_server_ip => #{new_ip[:target]} }, :basic_auth => #{basic_auth}"
```

Comment:
```suggestion
    $logger.error "Unable to set #{new_ip[:target]} as new active server ip for #{failover_ip} via POST to #{base_url}/failover/#{failover_ip}"
```

Reply:
Your formulation is better, however if you concur, I'd prefer to have `:body => { :active_server_ip => #{new_ip[:target]} }, :basic_auth => #{basic_auth}` be *somehow* displayed in the log. The reason is, when setting up Hetzner Robot access I often do manage to get it wrong the first time. So it's useful if I see all the params in the log and ... oh, this setting is wrong, gosh!

What do you think?

Reply:
i don't like so much a) including ruby specific data structures (the hash) as well as b) wrting the credentials in the log line. I'd rather add something like `..., payload: #{JSON.generate(body: { active_server_ip: { new_ip[:target] }})` if you think that's helpful, ... even though we then have to add json as explicit dependency of heartbeat.

Reply:
I concur with both of your points. I do not think we need to add any json to the output. I propose 53f0194